### DISCUSSION TITLE
Enhancement Request: Add links to outside URLs in Collections & Pages Blocks

### DISCUSSION THREAD

**Discussion by writingpoli** [knowledgefutures/pubpub#1232] [Ideas]
I'd like to be able to add links to external sites on my pages. Maybe something along the lines of [fyi.to](https://sareview.fyi.to)?  

![Screenshot of Enhancement Request from WP at SAR](https://f002.backblazeb2.com/file/WPSHAREX/ShareX/2021/01/pubpubcollectionblock-addlinksasimages.png)

SOLVES: 

Rebuilding page in HTML Block to include image links to external sites. 

EXAMPLE:

For instance, instead of using [our links page on fyi.to](https://sareview.fyi.to/sar-links) for our social profile links landing page, I could use [our page on PubPub](http://sareview.org/social#links) -- if it were easier to include external links in the

**Comment by writingpoli** [comment]
Collections & Pages Block in the Page Editor. 

(Github wouldn't let me edit the above when it chopped off the end during save.)

**Comment by gabestein** [comment]
Thanks! I really like this idea.

In an ideal world, do you think this is part of the pages/collection block, or a new type of block that maybe combines all types of PubPub objects, including external links? I guess one question I'm getting at is, should we give up the Pub/Page block distinction and just create a list block that allows you to combine different types of things, with some different display options? Or is there use to keeping those distinctions as well?

**Reply by writingpoli** [reply]
Given the format of those blocks, it really seems like a natural solution -- if it's easy-ish to implement. Having the URL pull the favicon or logo as the image the way fyi.to (and airtable and others) does would completely obviate my need to use another service for the purpose. :)

**Reply by jeffpooley** [reply]
I would make a plug for allowing all images to have an optional external link—not just in Page + Collection Blocks but in Pubs too, with any image added through the Media UI control. This would instantly expand the practical design flexibility across PubPub.

**Comment by writingpoli** [comment]
@gabestein You're probably right -- we could probably do without the distinction. I was just trying to be specific in my request using existing module names. 

I guess we'd have to call that block _something_.  :)

**Reply by gabestein** [reply]
@writingpoli Makes sense. And I think it's an open question! Certainly, it's harder to write queries across Pubs/Pages/Collections on our end than it is to keep them separate, and there are definitely some good reasons (different metadata, etc.) to keep them separate. But one thing we're pondering anyway is slowly combining the architecture of all of those things into one, which would point in this direction...so, all to say, worth thinking about.
